### PR TITLE
Fix DenseGeneral bug

### PR DESCRIPTION
**Pull request by jheek**
- Correctly expand bias to account
for batch & non-contracting dims.
- Sort axes for consistent output under permutations

### CODE REVIEW

File: tests/linen/linen_linear_test.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def _counter_init(rng, shape, dtype, state):
     np.testing.assert_allclose(y, target)
 
   @parameterized.parameters([((-2, 3), (), 'bijk,jklm->bilm'),
-                             ((3, -2), (), 'bijk,kjlm->bilm'),
+                             ((3, -2), (), 'bijk,jklm->bilm'),
```
Comment (marcvanzee) [inline tests/linen/linen_linear_test.py:147]: So our testcase was actually testing wrong behavior? That's odd 😄

File: tests/linen/linen_linear_test.py [hunk 2]
```diff
@@ -271,6 +271,18 @@ def test_embed(self):
     np.testing.assert_allclose(y, dummy_embedding[None])
     z = embed_module.apply(initial_params, jnp.ones((3,)), method=embed_module.attend)
     np.testing.assert_allclose(z, 3. * jnp.arange(4))
+  
+  def test_non_final_axis(self):
```
Comment (marcvanzee) [inline tests/linen/linen_linear_test.py:275]: For completeness we could also add a testcase where `axis` is a tuple, so we make sure it works on multiple axes as well.

**Review by marcvanzee — APPROVED** [review]

**Review by levskaya — APPROVED** [review]

File: tests/linen/linen_linear_test.py (hunk 1 above)
Comment (jheek) [reply tests/linen/linen_linear_test.py:147]: No this was correct. We are free to pick the order of the kernel dimensions. But I made it invariant wrt to the order in which you pass the axes. I think this is more consistent and it makes the implementation easier.

File: tests/linen/linen_linear_test.py (hunk 2 above)
Comment (jheek) [reply tests/linen/linen_linear_test.py:275]: done
